site/app/controllers/AbstractController.php

```diff
@@ -280,4 +281,35 @@ protected function tryGetAutocheck(AutoGradedTestcase $testcase, string $autoche
             return false;
         }
     }
+
+    /**
+     * Gets a course material from its path.
+     * @param string $path
+     * @param bool $render_json true to render a JSEND response to the output in the failure/error case
+     * @return CourseMaterial|bool false in the fail/error case
+     */
+    public function tryGetCourseMaterial(string $path, bool $render_json = true) {
+        if ($path === '') {
```

Comment:
You could use `file_exists` here to see if the path points to a valid file/dir which would be a stronger check and probably more common that someone passes a broken path then an empty string